Terraform files for provisioning Rabbit MQ

warrenbailey opened the pull request:
**What**
This pull requests does the following:
1. Creates two amazon instances
2. Reconfigurez the /etc/hosts and /etc/hostname files and set one to RabbitMQ1 and the other to RabbitMQ2
3. Creates two Route 53 entries; rabbitmq1.eq.ons.digital and rabbitmq2.eq.ons.digital
4. Executes an Ansible playbook to provision the two instances using the eq-messaging git hub repository.

**How to test**
1. Follow the steps in the README to set up terraform. (Essentially create a terraform.tfvars file using the example file, copy in the pre-prod key pair and install git and ansible).
2. Execute `terraform plan` and then `terraform apply`
3. Finally clean up using `terraform destroy`

**Who can review**
Anyone apart from @warrenbailey

warrenbailey committed:
Terraform initial check in - provision two ec2 instances and reconfigure the host names to rabbitmq1 and rabbitmq2

warrenbailey committed:
Invoke Ansible to provision the RabbitMQ instances and also set up the Route 53 information

warrenbailey committed:
Updated README

dhilton commented on README.md:
```diff
@@ -1 +1,27 @@
+2. Ensure that you have signed up for a heroku account and have generated an API
+key.
+
+3. Copy `terraform.tfvars.example` to `terraform.tfvars` and replace your heroku
```
I don't think we need the Heroku api key do we?

warrenbailey replied:
Good point, copy and paste issue there

dhilton commented on terraform.tfvars.example:
```diff
@@ -0,0 +1,3 @@
+aws_access_key="XXXXXXXXXXXX"
+aws_secret_key="XXXXXXXXXXXX"
+aws_key_pair="digital-eq-keypair"
```
Is this key pair used by this deployment? The referenced keypair above is `pre-prod.pem` ?

warrenbailey replied:
Yep you're right, should be pre-prod.pem. I'll change it

collisdigital commented:
Testing Slack integration again...

warrenbailey committed:
Updating README and example as per PR request comments

warrenbailey committed:
Adding terraform env name variable and updating readme with trouble shooting help

dhilton committed:
Hosts file now env aware

dhilton committed:
Refactor to minimise duplicate code

- Changed how the bash remote-exec's worked to reduce the need for templates and local preperation of config files first.
- Used indexes & count to reduce the code required to create the Rabbitmq servers.
- Added '-p' option to mkdir to fix error when a terraform run fails.
- Added a way to pass the deployment environment name to ansible via a command line flag

dhilton committed:
Change dns name to match correct pattern

dhilton committed:
Amend to support multiline bash shell redirection

dhilton commented:
@warrenbailey I've refactored the deployment code a little, making the bash parts a bit more succinct and removed the need for the local template dance. 

I've pushed a new branch to the ansible eq-messaging repo to fix the variables issue - the last part is the hosts file not being correctly set in this terraform run; the bash fails on multilines but the last commit should address that.

Amended branch: https://github.com/ONSdigital/eq-messaging/tree/EQ-18-Amend-variables

warrenbailey commented:
@dhilton that's looks a lot better. I didn't like that local file stuff, this is much better :+1:

warrenbailey committed:
Reverting to using a temporary hosts file

warrenbailey committed:
Adding environment variable to the hosts file

warrenbailey commented:
Tested and working.

dhilton commented:
LGTM :+1:
